connector/MongoDBAtlasODBC.pq

```diff
@@ -1,33 +1,74 @@
-﻿// This file contains your Data Connector logic
-[Version = "0.0.0"]
+﻿[Version = "0.0.0"]
 section MongoDBAtlasODBC;
 
 [DataSource.Kind="MongoDBAtlasODBC", Publish="MongoDBAtlasODBC.Publish"]
-shared MongoDBAtlasODBC.Contents = (optional message as text) =>
+shared MongoDBAtlasODBC.Contents = Value.ReplaceType(MongoDBAtlasODBCImpl, MongoDBAtlasODBCType);
+
+MongoDBAtlasODBCType = type function(
+    mongodbUri as (type text meta[
+        Documentation.FieldCaption = "MongoDB URI",
+        Documentation.FieldDescription = "MongoDB connection URI",
+        Documentation.SampleValues = {"mongodb://hostname:port/admin"}
+    ]), 
+    database as (type text meta [
+        Documentation.FieldCaption = "Database",
+        Documentation.FieldDescription = "Database to connect to",
+        Documentation.SampleValues = {"db"}
+    ]), 
+    optional options as record)
+    as table meta [
+        Documentation.Name = "MongoDB Atlas SQL"
+    ];
+
+MongoDBAtlasODBCImpl = (mongodbUri as text, database as text, optional options as record) as table => 
+let
+    Uri = ProcessUri(mongodbUri),
+    ConnectionString = [
+        Driver = "ADF_ODBC_DRIVER"
+    ],
+    CredentialConnectionString = GetCredentialConnectionString(),
+    OdbcDatasource = Odbc.DataSource(ConnectionString, [
+        // SQL-1118: Implement Odbc.Datasource SqlCapabilities and Extensible Options
+    ] & CredentialConnectionString)
+in
+    OdbcDatasource;
+
+GetCredentialConnectionString = () as record =>
     let
-        _message = if (message <> null) then message else "(no message)",
-        a = "Hello from MongoDBAtlasODBC: " & _message
+        Credential = Extension.CurrentCredential(),
+        EncryptionEnabled = Credential[EncryptConnection]? = true,

```

Comment:
Will this conflict with the SSL options users can set in the URI?
https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.ssl

Reply:
That's a good point.  I will remove the encryption handling here and have it handled in the ProcessUri only so it does not conflict and add a note to validate the encryption is behaving as expected.